modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/upgrade/v3_1_0/UrlSubjectUpgradeProcess.java

```diff
@@ -89,22 +63,33 @@ private String _getUrlSubject(long id, String subject) {
 
 	private void _populateUrlSubject() throws Exception {
 		try (PreparedStatement ps1 = connection.prepareStatement(
-				"select messageId, subject from MBMessage where (urlSubject " +
-					"is null) or (urlSubject = '')");
+				"select messageId, subject from MBMessage order by subject, " +
+					"messageId asc");
 			ResultSet rs = ps1.executeQuery();
 			PreparedStatement ps2 = AutoBatchPreparedStatementUtil.autoBatch(
 				connection.prepareStatement(
 					"update MBMessage set urlSubject = ? where messageId = " +
 						"?"))) {
 
+			int count = 0;
+			String currUrlSubject;
```

Comment:
Rename these to curURLSubject and previousURLSubject. Also, assign both to `null`.